lib/sdotypes.c

```diff
@@ -2978,10 +3040,25 @@ int sdo_rendezvous_list_add(sdo_rendezvous_list_t *list, sdo_rendezvous_t *rv)
 		// Now the enty_ptr is pointing to the last entry
 		// Add the r entry onto the end
 		prev_ptr->next = rv;
-		list->num_entries++;
+		directives->num_entries++;
 	}
-	LOG(LOG_DEBUG, "Added to rvlst, %d entries\n", list->num_entries);
-	return list->num_entries;
+	LOG(LOG_DEBUG, "Added to rvlst, %d entries\n", directives->num_entries);
+	return directives->num_entries;
+}
+
+sdo_rendezvous_directive_t *sdo_rendezvous_directive_get(sdo_rendezvous_list_t *list, int num)
+{
+	int index;
+
+	if (list == NULL || list->rv_directives == NULL)
+		return NULL;
+
+	sdo_rendezvous_directive_t *entry_ptr = list->rv_directives;
+
+	for (index = 0; index < num; index++) {
+		entry_ptr = entry_ptr->next;
```

Comment:
Possibility of null pointer dereference if num is more than the number of RV Directives. Should put a check for `entry_ptr->next` before assigning.

Reply:
Updated.